shared/src/main/java/com/couchbase/lite/AbstractReplicatorConfiguration.java

```diff
@@ -0,0 +1,405 @@
+//
+// ReplicatorConfiguration.java
+//
+// Copyright (c) 2017 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package com.couchbase.lite;
+
+import android.support.annotation.NonNull;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import com.couchbase.lite.internal.core.CBLVersion;
+
+
+/**
+ * Replicator configuration.
+ */
+public abstract class AbstractReplicatorConfiguration {
```

Comment:
This class shouldn't be public. I understand that Java allows to override non-public abstract class to hide the class from public visibility.

Reply:
It has to be public.  All those constants are public.  It would be insanity to define the multiple times (as used to be the case).  What is the problem with making it public.  All of its contents used to be in a public class?

Reply:
I think that would work.